tools/integration_tests/gzip/read_gzip_test.go

```diff
@@ -0,0 +1,167 @@
+	}
+}
+
+// Verify that the passed file exists on the GCS test-bucket and in the mounted bucket
+// and its full read returns the same size as the size of the
+// GCS object.
+func genericFullFileReadTest(t *testing.T, filename string) {
+	mountedFilePath := path.Join(setup.MntDir(), TestBucketPrefixPath, filename)
+	data, err := operations.ReadFile(mountedFilePath)
+	if err != nil {
+		t.Errorf("Failed to read mounted file %s: %v", mountedFilePath, err)
+	}
+
+	gcsObjectPath := path.Join(setup.TestBucket(), TestBucketPrefixPath, filename)
+	gcsObjectSize, err := operations.GetGcsObjectSize(gcsObjectPath)
+	if err != nil {
+		t.Fatalf("Failed to get size of gcs object %s: %v\n", gcsObjectPath, err)
+	}
+	if len(data) != gcsObjectSize {
+		t.Fatalf("Read size for %s (%d bytes) didn't match the size of the GCS object gs://%s (%d bytes)", mountedFilePath, len(data), gcsObjectPath, gcsObjectSize)
+	}
+	if strings.Contains(string(data), helpers.TempFileStrLine) {
+		t.Fatalf("Read data contains '%s' but shouldn't", helpers.TempFileStrLine)
+	}
+}
+
+// Verify that the passed file exists on the GCS test-bucket and in the mounted bucket
+// and its ranged read returns the same size as the requested read size.
+func genericRangedReadTest(t *testing.T, filename string) {
+	mountedFilePath := path.Join(setup.MntDir(), TestBucketPrefixPath, filename)
+
+	gcsObjectPath := path.Join(setup.TestBucket(), TestBucketPrefixPath, filename)
+	gcsObjectSize, err := operations.GetGcsObjectSize(gcsObjectPath)
+	if err != nil {
+		t.Fatalf("Failed to get size of gcs object %s: %v\n", gcsObjectPath, err)
+	}
+
+	readSize := gcsObjectSize / 10
+	readOffset := readSize / 10
+	f, err := os.Open(mountedFilePath)
+	if err != nil || f == nil {
+		t.Fatalf("Failed to open local mounted file %s: %v", mountedFilePath, err)
+	}
+
+	buf := make([]byte, readSize)
+	for _, offsetMultiplier := range []int64{1, 3, 5, 7, 9} {
+		n, err := f.ReadAt(buf, offsetMultiplier*int64(readOffset))
+		if err != nil {
+			t.Fatalf("Failed to read mounted file %s: %v", mountedFilePath, err)
+		}
+
+		if readSize != n {
+			t.Fatalf("Read size for %s (%d bytes) didn't match the size of the requested read-size (%d bytes)", mountedFilePath, n, readSize)
+		}
+
+		if strings.Contains(string(buf), helpers.TempFileStrLine) {
+			t.Fatalf("Read buffer contains '%s' but shouldn't", helpers.TempFileStrLine)
+		}
+	}
+}
+
+func TestGzipEncodedTextFileWithNoTransformSize(t *testing.T) {
+	genericFileSizeTest(t, TextContentWithContentEncodingWithNoTransformFilename)
+}
+
+func TestGzipEncodedTextFileWithNoTransformFullRead(t *testing.T) {
+	genericFullFileReadTest(t, TextContentWithContentEncodingWithNoTransformFilename)
+}
+
+func TestGzipEncodedTextFileWithNoTransformRangedRead(t *testing.T) {
+	genericRangedReadTest(t, TextContentWithContentEncodingWithNoTransformFilename)
+}
+
+func TestGzipEncodedTextFileWithoutNoTransformSize(t *testing.T) {
```

Comment:
I purposefully kept those two as separate tests because of
1. For granularity of testing, and
2. The size test checks the output of stat call or `os.Stat` as opposed to the read size which tests the NewReader call.

There is of course redundancy between the two, so I'll merge the two tests into one.

Reply:
Merged into one.